ReactWindows/ReactNative.Shared/Bridge/ReactInstance.cs

```diff
@@ -142,23 +138,40 @@ public void InvokeCallback(int callbackId, JArray arguments)
 
         public /* TODO: internal? */ void InvokeFunction(string module, string method, JArray arguments, string tracingName)
         {
+            DispatchBridgeAction(() =>
+            {
+                using (Tracer.Trace(Tracer.TRACE_TAG_REACT_BRIDGE, tracingName).Start())
+                {
+                    _bridge.CallFunction(module, method, arguments);
+                }
+            });
+        }
+
+        private void DispatchBridgeAction(Action action)
+        {
+            if (IsDisposed)
+            {
+                Tracer.Write(ReactConstants.Tag, "Ignoring Invocation of JS function/callback as the bridge has been destroyed.");
+                return;
+            }
+
             QueueConfiguration.JavaScriptQueue.Dispatch(() =>
             {
                 QueueConfiguration.JavaScriptQueue.AssertOnThread();
 
                 if (IsDisposed)
                 {
+                    Tracer.Write(ReactConstants.Tag, "Ignoring Invocation of JS function/callback as the bridge has been destroyed.");
                     return;
                 }
 
-                using (Tracer.Trace(Tracer.TRACE_TAG_REACT_BRIDGE, tracingName).Start())
+                if (_bridge == null)
```

Comment:
I think this if check can be moved out of this block and the stashing can be done directly from the caller thread instead of on JavaScriptQueue. But the earlier if check was in the JavaScriptQueue and since I am not sure why it was that way, I kept this check inside too. If someone who know the design better can confirm that it is safe to move this if block outside, I will make that change.